# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.cpp
# Commit: c420936bf587
# Full Hash: c420936bf587e0ebe62d59f618bc4fefe5ea57df
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-24 21:44:48
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Add logging in all branches of AudioSinkWrapper. r=alwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D146197
# ==============================================================================

diff -r 628ab886c5ec -r c420936bf587 dom/media/mediasink/AudioSinkWrapper.cpp
--- a/dom/media/mediasink/AudioSinkWrapper.cpp	Tue May 24 13:09:10 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.cpp	Tue May 24 13:09:11 2022 +0000
@@ -148,7 +148,8 @@
     audio = mAudioQueue.PopFront();
     dropped++;
     if (audio) {
-      LOG("Dropping audio packets: media position: %lf, "
+      LOGV(
+          "Dropping audio packets: media position: %lf, "
           "packet dropped: [%lf, %lf] (%u so far).\n",
           aMediaPosition.ToSeconds(), audio->mTime.ToSeconds(),
           (audio->mTime + audio->mDuration).ToSeconds(), dropped);
@@ -181,9 +182,12 @@
     }
   } else {
     if (!IsPlaying()) {
+      LOG("%p: AudioSinkWrapper::OnMuted: not playing, not re-creating an "
+          "AudioSink",
+          this);
       return;
     }
-    LOG("AudioSinkWrapper unmuted, re-creating an AudioStream.");
+    LOG("%p: AudioSinkWrapper unmuted, re-creating an AudioStream.", this);
     TimeUnit mediaPosition = GetSystemClockPosition(TimeStamp::Now());
     nsresult rv = StartAudioSink(mediaPosition, AudioSinkStartPolicy::ASYNC);
     if (NS_FAILED(rv)) {
@@ -248,7 +252,7 @@
 
 void AudioSinkWrapper::SetPlaying(bool aPlaying) {
   AssertOwnerThread();
-  LOG("%p: SetPlaying %s", this, aPlaying ? "true" : "false");
+  LOG("%p: AudioSinkWrapper::SetPlaying %s", this, aPlaying ? "true" : "false");
 
   // Resume/pause matters only when playback started.
   if (!mIsStarted) {
@@ -285,6 +289,7 @@
 
 nsresult AudioSinkWrapper::Start(const TimeUnit& aStartTime,
                                  const MediaInfo& aInfo) {
+  LOG("%p AudioSinkWrapper::Start", this);
   AssertOwnerThread();
   MOZ_ASSERT(!mIsStarted, "playback already started.");
 
@@ -318,14 +323,14 @@
              &AudioSinkWrapper::OnAudioEnded, &AudioSinkWrapper::OnAudioEnded)
       ->Track(mAudioSinkEndedPromise);
 
-  LOG("AudioSinkWrapper::StartAudioSink (%s)",
+  LOG("%p: AudioSinkWrapper::StartAudioSink (%s)", this,
       aPolicy == AudioSinkStartPolicy::ASYNC ? "Async" : "Sync");
 
   if (IsMuted()) {
-    LOG("Muted: not starting an audio sink");
+    LOG("%p: Muted: not starting an audio sink", this);
     return NS_OK;
   }
-  LOG("Not muted: starting a new audio sink");
+  LOG("%p: Not muted: starting a new audio sink", this);
   if (aPolicy == AudioSinkStartPolicy::ASYNC) {
     UniquePtr<AudioSink> audioSink;
     audioSink.reset(mCreator->Create());
@@ -417,7 +422,7 @@
   AssertOwnerThread();
   MOZ_ASSERT(mIsStarted, "playback not started.");
 
-  LOG("%p: Stop", this);
+  LOG("%p: AudioSinkWrapper::Stop", this);
 
   mIsStarted = false;
   mAudioEnded = true;
@@ -445,6 +450,7 @@
 
 void AudioSinkWrapper::OnAudioEnded() {
   AssertOwnerThread();
+  LOG("%p: AudioSinkWrapper::OnAudioEnded", this);
   mAudioSinkEndedPromise.Complete();
   mPlayDuration = GetPosition();
   if (!mPlayStartTime.IsNull()) {
